griptape/loaders/email_loader.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import annotations
+from attr import astuple, define, field
+from griptape import utils
+from griptape.artifacts import ErrorArtifact, ListArtifact, TextArtifact
+from griptape.loaders import BaseLoader
+from typing import Optional
+import imaplib
+import logging
+import mailparser
+
+
+@define(frozen=True)
+class EmailQuery:
+    """An email retrieval query
+
+    Attributes:
+        label: Label to retrieve emails from such as 'INBOX' or 'SENT'.
+        key: Optional key for filtering such as 'FROM' or 'SUBJECT'.
+        search_criteria: Optional search criteria to filter emails by key.
+        max_count: Optional max email count.
+    """
+
+    label: str = field(kw_only=True)
+    key: Optional[str] = field(default=None, kw_only=True)
+    search_criteria: Optional[str] = field(default=None, kw_only=True)
+    max_count: Optional[int] = field(default=None, kw_only=True)
+
+@define
+class EmailLoader(BaseLoader):
+    url: str = field(kw_only=True)
```

Comment:
Could be potentially unclear what URL/protocol is used. Can we rename this to `imap_url`, so it's more descriptive?